types/stake.go

```diff
@@ -67,14 +68,15 @@ type ValidatorSet interface {
 	IterateValidatorsBonded(Context,
 		func(index int64, validator Validator) (stop bool))
 
-	Validator(Context, ValAddress) Validator            // get a particular validator by operator
-	ValidatorByPubKey(Context, crypto.PubKey) Validator // get a particular validator by signing PubKey
-	TotalPower(Context) Dec                             // total power of the validator set
+	Validator(Context, ValAddress) Validator                // get a particular validator by operator address
+	ValidatorByConsPubKey(Context, crypto.PubKey) Validator // get a particular validator by consensus PubKey
```

Comment:
Should we get rid of this method now, since we no longer keep the pubkey index?

Reply:
Okay - well we no longer keep the pubkey index - but in certain situations, all you are going to have access to is the pubkey - so the idea was to have it be a bit nicer than having to constantly type `sdk.ConsAddress(pubKey.Address())`  - BUT I suppose this can be added to a new helper function in `types/` maybe named `GetConsAddr(pubkey)` - If we do that then I'm game to remove the `ValidatorByConsPubKey`. Thoughts?

Reply:
I think we should do this @rigelrozanski